src/plugins/dna_export/src/ExportSequenceTask.cpp

```diff
@@ -553,18 +553,31 @@ U2Sequence ExportAnnotationSequenceSubTask::importAnnotatedSeq2Dbi(const SharedA
     importer.startSequence(os, resultDbiRef, U2ObjectDbi::ROOT_FOLDER, ad->name, false);
     CHECK_OP(os, U2Sequence());
 
-    foreach (const U2Region &annotatedRegion, ad->location->regions) {
+    bool isReverseComplement = ad->getStrand().isCompementary() && nullptr != ei.complTT;
+    for (const U2Region &annotatedRegion : qAsConst(ad->location->regions)) {
         qint64 currentRegionLength = 0;
-        for (qint64 pos = annotatedRegion.startPos; pos < annotatedRegion.endPos(); pos += MAX_CHUNK_LENGTH) {
-            const qint64 currentChunkSize = qMin(MAX_CHUNK_LENGTH, annotatedRegion.endPos() - pos);
+        QList<QByteArray> chunkContentList;
+        // If we need to export a reverse-complementary sequence, this sequence should be translated to amino acid,
+        // and sequence length isn't divided by 3, then we need to skip one or two characters from the beginning -
+        // owervise the translation wiil be shifted on this not-skipped characters.
+        qint64 offset = 0;
+        if (isReverseComplement) {
+            offset = annotatedRegion.length % 3;
+        }
+        qint64 currentChunkSize = MAX_CHUNK_LENGTH;
+        for (qint64 pos = annotatedRegion.startPos; pos < annotatedRegion.endPos(); pos += currentChunkSize) {
+            currentChunkSize = qMin(MAX_CHUNK_LENGTH, annotatedRegion.endPos() - pos);
+            if (chunkContentList.isEmpty() && offset != 0) {
+                currentChunkSize = offset;
```

Comment:
offset is always between 0 and 2: the chunk size of length <=2 can't be translated, so the code when currentChunkSize= offset is no-op. Why not start with the `pos = annotatedRegion.startPos+offset` to avoid the complexity inside of the loop?

Reply:
You lose first 1 or 2 bases